it/src/main/java/org/corfudb/universe/universe/docker/FakeDns.java

```diff
@@ -0,0 +1,187 @@
+        return instance;
+    }
+
+    public synchronized void addForwardResolution(String hostname, InetAddress ip) {
+        forwardResolutions.put(hostname, ip);
+    }
+
+    public synchronized void addReverseResolution(InetAddress ip, String hostname) {
+        reverseResolutions.put(ip, hostname);
+    }
+
+    /**
+     * Install the fake DNS resolver into the Java runtime.
+     */
+    public synchronized FakeDns install() {
+        if (installed) return this;
+        try {
+            installDns();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+        installed = true;
+
+        return this;
+    }
+
+    private void installDns() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException,
+            ClassNotFoundException, NoSuchFieldException {
+        try {
+            // Override the NameService in Java 9 or later.
+            final Class<?> nameServiceInterface = Class.forName("java.net.InetAddress$NameService");
+            Field field = InetAddress.class.getDeclaredField("nameService");
+            // Get the default NameService to fallback to.
+            Method method = InetAddress.class.getDeclaredMethod("createNameService");
+            method.setAccessible(true);
+            Object fallbackNameService = method.invoke(null);
+            // Create a proxy instance to set on the InetAddress field which will handle
+            // all NameService calls.
+            Object proxy = Proxy.newProxyInstance(
+                    nameServiceInterface.getClassLoader(),
+                    new Class<?>[]{nameServiceInterface},
+                    new NameServiceListener(fallbackNameService)
+            );
+            field.setAccessible(true);
+            field.set(InetAddress.class, proxy);
+        } catch (ClassNotFoundException | NoSuchFieldException e) {
+            // Override the NameService in Java 8 or earlier.
+            final Class<?> nameServiceInterface = Class.forName("sun.net.spi.nameservice.NameService");
+            Field field = InetAddress.class.getDeclaredField("nameServices");
+            // Get the default NameService to fallback to.
+            Method method = InetAddress.class.getDeclaredMethod("createNSProvider", String.class);
+            method.setAccessible(true);
+            Object fallbackNameService = method.invoke(null, "default");
+            // Create a proxy instance to set on the InetAddress field which will handle
+            // all NameService calls.
+            Object proxy = Proxy.newProxyInstance(
+                    nameServiceInterface.getClassLoader(),
+                    new Class<?>[]{nameServiceInterface},
+                    new NameServiceListener(fallbackNameService)
+            );
+            field.setAccessible(true);
+            // Java 8 or earlier takes a list of NameServices
+            field.set(InetAddress.class, Arrays.asList(proxy));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure that this accessibility bypass is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3011)